crates/bevy_core/src/time/time.rs

```diff
@@ -33,115 +56,411 @@ impl Time {
+    /// The exact CPU time since the last update, as [`f32`] seconds.
+    /// **Not** affected by [`Time::relative_speed`].
     #[inline]
-    pub fn startup(&self) -> Instant {
-        self.startup
+    pub fn raw_delta_seconds(&self) -> f32 {
+        self.raw_delta_seconds
+    }
+
+    /// The exact CPU time since the last update, as [`f64`] seconds.
+    /// **Not** affected by [`Time::relative_speed`].
+    #[inline]
+    pub fn raw_delta_seconds_f64(&self) -> f64 {
+        self.raw_delta_seconds_f64
+    }
+
+    /// The reciprocal of the rate set by [`Time::set_steps_per_second`], as a [`Duration`].
+    ///
+    /// A fixed timestep only guarantees that N steps run for each second of elapsed time, not that
+    /// one step runs every [`Time::fixed_delta`] seconds. The exact CPU time between steps depends
+    /// on the frame rate, so use [`Time::fixed_delta`] in systems for consistent behavior.
+    #[inline]
+    pub fn fixed_delta(&self) -> Duration {
+        self.fixed_delta
+    }
+
+    /// The reciprocal of the rate set by [`Time::set_steps_per_second`], as [`f32`] seconds.
+    ///
+    /// A fixed timestep only guarantees that N steps run for each second of elapsed time, not that
+    /// one step runs every [`Time::fixed_delta`] seconds. The exact CPU time between steps depends
+    /// on the frame rate, so use [`Time::fixed_delta`] in systems for consistent behavior.
+    #[inline]
+    pub fn fixed_delta_seconds(&self) -> f32 {
+        self.fixed_delta_seconds
+    }
+
+    /// The reciprocal of the rate set by [`Time::set_steps_per_second`], as [`f64`] seconds.
+    ///
+    /// A fixed timestep only guarantees that N steps run for each second of elapsed time, not that
+    /// one step runs every [`Time::fixed_delta`] seconds. The exact CPU time between steps depends
+    /// on the frame rate, so use [`Time::fixed_delta`] in systems for consistent behavior.
+    #[inline]
+    pub fn fixed_delta_seconds_f64(&self) -> f64 {
+        self.fixed_delta_seconds_f64
+    }
+
+    /// Change the step Hz (changes [`Time::fixed_delta`]). Users are strongly cautioned against
+    /// changing this outside of startup. Use [`Time::set_relative_speed`] to ensure that systems
+    /// using [`fixed_delta`] behave consistently.
+    pub fn set_steps_per_second(&mut self, rate: f32) {
+        self.fixed_delta = Duration::from_secs_f32(1.0 / rate);
+        self.fixed_delta_seconds = self.fixed_delta.as_secs_f32();
+        self.fixed_delta_seconds_f64 = self.fixed_delta.as_secs_f64();
     }
 
-    /// The ['Instant'] when [`Time::update`] was last called, if it exists
+    /// The rate that time advances relative to CPU time, as [`f32`]. 1.0 by default.
+    #[inline]
+    pub fn relative_speed(&self) -> f32 {
+        self.relative_speed
+    }
+
+    /// The rate that time advances relative to CPU time, as [`f64`]. 1.0 by default.
+    #[inline]
+    pub fn relative_speed_f64(&self) -> f64 {
```

Comment:
Is this necessary? I could see doing things to allow a future migration to `f64` backing field but I doubt the accuracy (and certainly not the range) would be helpful here.

Reply:
Hmm, good question. 

I'm wondering if there should be any `f64` convenience methods at all. The precision is a bit inconsistent as-is since `duration.as_secs_f32() as f64` and `duration.as_secs_f64()` aren't equivalent, so maybe the float parts should just be 32-bit.

Reply:
Removed the `f64` methods that were just plain casts, kept them for the `Duration`-based values.